subsys/net/ip/tcp2.c

```diff
@@ -1070,6 +1075,9 @@ static void tcp_resend_data(struct k_work *work)
 
 			goto out;
 		}
+	} else if (ret == -ENODATA) {
+		conn->data_mode = TCP_DATA_MODE_SEND;
+		goto out;
```

Comment:
> when we detect that retransmission is no longer needed, it should be skipped I believe.

I add this code so that no more retransmission if no data to send.

The following is debug message after this patch is applied. tcp_send_data() is not repeatedly called when no data to send. 
```
tcp_in: conn=0x567e64b8 send_data_total=0 unacked_len=0 send_timer.work.flags=0x104(queued) uptime=4380
tcp_send_data: conn=0x567e64b8 no data to send, thread='tcp_work', uptime=4380
tcp_in: conn=0x567e69dc send_data_total=0 unacked_len=0 send_timer.work.flags=0x104(queued) uptime=4740
tcp_send_data: conn=0x567e69dc no data to send, thread='tcp_work', uptime=4740
```